src/sidebar/components/annotation-action-bar.js

```diff
@@ -53,7 +53,7 @@ function AnnotationActionBar({
   //  Only authenticated users can flag an annotation, except the annotation's author.
   const showFlagAction =
     !!userProfile.userid && userProfile.userid !== annotation.user;
-  const showShareAction = isShareable(annotation, settings);
+  const showShareAction = sharingEnabled(settings) && shareURI(annotation);
```

Comment:
The overall refactor of `annotation-sharing` logic made it more coherent to have this logic in two discrete functions instead of one. There is some complexity because "is sharing allowed" is not a single question, but, kind of, four:

* Should we show the icon for sharing this page's annotations? The logic we use (unchanged) is "Are we first-party?" This is in `AnnotationActionBar` at present.
* Should we show the icon for sharing a single annotation? (This line here). The logic we use (unchanged) is: is there anything in the app's config that says we _shouldn't_ allow annotation sharing, and does this annotation have a valid single-annotation URL in its `links`?
* Which UI should we be providing within the "share all annotations on this page" panel, once it's open? (New). If the page's URI is shareable (i.e. web-accessible), we render the main, pre-existing UI that provides a bouncer link. If not, we render the new UI path explaining why the page's annotations are not shareable (with no link).
* Which UI should we be providing within the "share this annotation" panel, once it's open? (New). If the annotation's URI is shareable (i.e. web-accessible), we render the main, pre-existing UI that provides a bouncer link. If not, we render the new UI path explaining why this annotation is not shareable (with no link).